spec/requests/medications_spec.rb

```diff
@@ -144,7 +95,7 @@ def post_create(medication_params)
           it 'has no validation errors and has correct refill' do
             post_create valid_medication_params
             expect(assigns(:medication).errors).to be_empty
-            expect(assigns(:medication)[:refill].strftime('%d/%m/%Y')).to eq('01/01/2020')
+            expect(response.body).to include(valid_medication_params.refill)
           end
         end
 
```

Comment:
At a bit of a loss on this one. Have several thoughts, none of which seems obviously best to me:

1. calculate from and to dates using `valid_medication_params` field values. Generate list of dates spanning the two dates. Check that at least one date on the list is included in the `response.body`.

2. make it a json request. Parse body for medication refill date. Calculate range as in 1. and check that refill falls in that range.

3. scrap all expectations. say responsibility for testing refill auto-gen lies with some method-level unit test instead. Here it is sufficient that the response successfully redirects to `medication_path(valid_medication_params.name`

4. something else entirely

Reply:
See my other comment.

I'd say maybe 3. I'm on mobile and haven't checked the controller code, but the responsibilites of the request spec should be to ensure the request passes and/or fails. It can be argued it would be good to check the output of  requests, especially for json requests (I'm in favor of having output checkes for json)

My belief is to think of these tests like black boxes. Check the in, and check the out. Ensure it succeeded or failed with the correct output. That's another reason we're removing all the `assigns` checks. That and those matchers are part of a gem that's being removed.

cc @bennpham